aea/skills/tasks.py

```diff
@@ -18,72 +18,102 @@
-from concurrent.futures import Executor, ThreadPoolExecutor
-from queue import Queue
-from threading import Thread
-from typing import Optional
-
-from aea.skills.base import Task
+from multiprocessing.pool import AsyncResult, Pool
+from typing import Any, Callable, Dict, Optional, Sequence, cast
 
 logger = logging.getLogger(__name__)
 
 
+def init_worker():
+    """
+    Initialize a worker.
+
+    Disable the SIGINT handler.
+    Related to a well-known bug: https://bugs.python.org/issue8296
+    """
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+
 class TaskManager:
     """A Task manager."""
 
-    def __init__(self, executor: Optional[Executor] = None):
+    def __init__(self, nb_workers: int = 5):
         """
         Initialize the task manager.
 
-        :param executor: the executor
+        :param nb_workers: the number of worker processes.
         """
-        self._executor = executor if executor is not None else ThreadPoolExecutor()
-
-        self.task_queue = Queue()  # type: Queue
-        self.futures = deque([])  # type: deque
-
-        self.stopped = True
-        self.thread = Thread(target=self.dispatch)
-        self.lock = threading.Lock()
-
-    def enqueue_task(self, task: Task) -> None:
+        self._nb_workers = nb_workers
+        self._pool = None  # type: Optional[Pool]
+        self._stopped = True
+        self._lock = threading.Lock()
+
+        self._task_enqueued_counter = 0
+        self._results_by_task_id = {}  # type: Dict[int, Any]
+
+    @property
+    def nb_workers(self) -> int:
+        """Get the number of workers."""
+        return self._nb_workers
+
+    def enqueue_task(
+        self, func: Callable, args: Sequence = (), kwds: Optional[Dict[str, Any]] = None
+    ) -> AsyncResult:
         """
         Enqueue a task with the executor.
 
-        :param task: the task instance to be enqueued
+        :param func: the callable instance to be enqueued
+        :param args: the positional arguments to be passed to the function.
+        :param kwds: the keyword arguments to be passed to the function.
+        :return the task id to get the the result.
+        :raises ValueError: if the task manager is not running.
         """
-        self.task_queue.put(task)
-
-    def dispatch(self) -> None:
-        """Dispatch a task."""
-        while not self.stopped:
-            for future in self.futures:
-                if future.done():
-                    future.result()
-
-            next_task = self.task_queue.get(block=True)  # type: Optional[Task]
-            if next_task is None:
-                return
-
-            future = self._executor.submit(next_task.execute)
-            self.futures.append(future)
+        with self._lock:
+            if self._stopped:
+                raise ValueError("Task manager not running.")
+            self._pool = cast(Pool, self._pool)
+            task_id = self._task_enqueued_counter
+            self._task_enqueued_counter += 1
```

Comment:
we assign a unique id to every submitted task. The consistency is guaranteed by the lock.

Reply:
Worth adding a comment in code to have this info maintained